packages/plugins/i18n/server/src/register.ts

```diff
@@ -61,6 +61,29 @@ const extendContentTypes = (strapi: Core.Strapi) => {
       visible: false,
       type: 'string',
     });
+
+    _.set(attributes, 'localizations', {
+      type: 'relation',
+      relation: 'oneToMany',
+      target: contentType.uid,
+      writable: false,
+      private: false,
+      configurable: false,
+      visible: false,
+      joinColumn: {
+        name: 'document_id',
+        referencedColumn: 'document_id',
+        referencedTable: strapi.db.metadata.identifiers.getTableName(contentType.collectionName!),
+        // ensure the population will not include the results we already loaded
+        on({ results }: { results: any[] }) {
+          return {
+            id: {
+              $notIn: results.map((r) => r.id),
+            },
+          };
+        },
```

Comment:
it actually doesn't return both because it's prefiltered with the work you did previously on relation to not populate different publication state so it applies to this too :)

Reply:
I dont think we propagate any status on the filters anymore 🤔

Reply:
Yes let me correct my sentence: The db relations are only draft to draft & published to published so it will always return the right stuff.

In the case where D&P is not enabled since it's a self relation there are no risks of D&P on one side & no D&P on the other it either D&P/D&P or noD&p/noD&P

Reply:
Maybe I misunderstood how this `localization` relation works 🤔 

Doesn't this joinColumn join by the "documentId" ? and hence return every result that matches the document same document id ?

Reply:
yeah you are right I mixed things up with normal relations 🤦 

 I'll check the code but doing manual testing I didn't get mixed up data 🤔

Reply:
@Marc-Roig  we still have `convertStatusParams` being applied in `transformQueryParams`  that does cascading that's why it works just fine. We needed it for noDp to DP etc so we re-added if you recall :)